api-1.9/src/main/java/org/openmrs/module/fhir/helper/ConditionHelperImpl1_9.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.fhir.helper;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.hl7.fhir.dstu3.model.Condition;
+import org.openmrs.Obs;
+import org.openmrs.annotation.OpenmrsProfile;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.fhir.api.helper.ConditionHelper;
+import org.openmrs.module.fhir.api.util.FHIRConditionUtil;
+import org.openmrs.module.fhir.api.util.FHIRUtils;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component(value = "fhir.ConditionHelper")
+@OpenmrsProfile(openmrsPlatformVersion = "1.9.*")
+public class ConditionHelperImpl1_9 implements ConditionHelper {
```

Comment:
Why do we need a 1.9.x impl as well as a 1.11.x impl? I get why 1.x versus 2.x would be a big deal, but not why 1.9 versus 1.11 would be a big deal.

Asking out of curiosity

Reply:
Very good point @joeldenning 👍

Reply:
@joeldenning @dkayiwa 
You are right. there's no difference between 1.11.x and 1.9.x but there's a huge difference in version 2.0.x and 2.2.x. 
That's why you find a class implementation for 1.11, 1.9 and 2.0 are just skeleton but the actual code is in the FHIRConditionUtil class.
This is how Allergy and Drug were implemented

Reply:
If there is no difference, why not just combine them into one class, with an openmrsPlatformVersion range, instead of duplicating the same exact code in two classes?

Reply:
<img width="212" alt="Screenshot 2019-10-31 at 08 37 39" src="https://user-images.githubusercontent.com/19473115/67921777-e3bf5700-fbb9-11e9-8cb1-dbd7009328a7.png">
 This will be great.
@dkayiwa  where do you suggest to put the new class (which combines 1.11, 1.9, 2.0), since the previous implementation has different API modules for different versions as you can see from the above image.
Will it be confusing to have a one class with openmrsPlatformVersion range `@OpenmrsProfile(openmrsPlatformVersion = "1.9.* - 2.1.*")`  at api-2.0? It will be inconsistent.

What you mean is the previous implementation wasn't done in the right way. 
compare this two implementation [FHIRAllergyIntoleranceUtil1_9.java](https://github.com/openmrs/openmrs-module-fhir/blob/master/api-1.9/src/main/java/org/openmrs/module/fhir/util/FHIRAllergyIntoleranceUtil1_9.java) and [FHIRAllergyIntoleranceUtil1_11.java](https://github.com/openmrs/openmrs-module-fhir/blob/master/api-1.11/src/main/java/org/openmrs/module/fhir/util/FHIRAllergyIntoleranceUtil1_11.java) . To me they are duplicate. Maybe there was a reason for this or none. I don't know

@dkayiwa need your advice on this